astroquery/eso/core.py

```diff
@@ -296,248 +218,472 @@ def _login(self, *, username: str = None, store_password: bool = False,
+            Authentication must be done beforehand via
+            :meth:`~astroquery.eso.EsoClass.login`. Authenticated queries may be slower.
+            Default is ``False``.
+        column_filters : dict or None, optional
+            Constraints applied to the query in ADQL syntax,
+            e.g., ``{"exp_start": "between '2024-12-31' and '2025-12-31'"}``.
+            Default is ``None``.
+        open_form : bool, optional
+            **Deprecated** - unused.
+        cache : bool, optional
+            **Deprecated** - unused.
 
         Returns
         -------
-        table : `~astropy.table.Table`
-            A table representing the data available in the archive for the
-            specified instrument, matching the constraints specified in
-            ``kwargs``. The number of rows returned is capped by the
-            ROW_LIMIT configuration item.
-
+        astropy.table.Table, str, int or None
+            - By default, returns an :class:`~astropy.table.Table` containing records
+              based on the specified columns and constraints.
+              Returns ``None`` if the query has no results.
+            - When ``count_only`` is ``True``, returns an ``int`` representing the
+              record count for the specified filters.
+            - When ``query_str_only`` is ``True``, returns the query string that
+              would be issued to the TAP service given the specified arguments.
         """
-        url = self.QUERY_INSTRUMENT_URL + "/eso_archive_main/form"
-        return self._query(url, column_filters=column_filters, columns=columns,
-                           open_form=open_form, help=help, cache=cache, **kwargs)
-
-    def query_instrument(self, instrument, *, column_filters={}, columns=[],
-                         open_form=False, help=False, cache=True, **kwargs):
+        _ = open_form, cache  # make explicit that we are aware these arguments are unused
+        column_filters = column_filters if column_filters else {}
+        up = _UserParams(table_name=_EsoNames.raw_table,
+                         column_name=_EsoNames.raw_instruments_column,
+                         allowed_values=instruments,
+                         cone_ra=cone_ra,
+                         cone_dec=cone_dec,
+                         cone_radius=cone_radius,
+                         columns=columns,
+                         column_filters=column_filters,
+                         top=top,
+                         count_only=count_only,
+                         query_str_only=query_str_only,
+                         print_help=help,
+                         authenticated=authenticated,
+                         )
+        t = self._query_on_allowed_values(up)
+        t = reorder_columns(t, DEFAULT_LEAD_COLS_RAW)
+        return t
+
+    @deprecated_renamed_argument(('open_form', 'cache'), (None, None),
+                                 since=['0.4.11', '0.4.11'])
+    def query_instrument(
```

Comment:
is there a need for this now as it feels to be just a special case for `query_main` above

Reply:
query_instruments and query_main query from two different data sources, so they are both needed.